Tests/RockLib.Logging.Tests/DependencyInjection/LoggerBuilderTests.cs

```diff
@@ -141,21 +150,29 @@ public void BuildMethodHappyPath1()
 
             logger.Should().BeOfType<Logger>()
                 .Which.LogProcessor.Should().BeSameAs(logProcessor);
-            logger.Level.Should().Be(LogLevel.Info);
+            logger.Level.Should().Be(customLogLevelResolver ? LogLevel.Error : LogLevel.Info);
```

Comment:
Instead of this check reading the value of `customLogLevelResolver` to determine what the `LogLevel` should be, just pass that in with `InlineData` - e.g. `[InlineData(true, LogLevel.Error)]`

Reply:
I can do that, but if it's the same value every time, what's the point in passing it in?

Also, the check is used to determine if the log level should be what the `ILogLevelResolver` return, or what the default `LogLevel` is. I purposely am keeping these different as a way to validate that the level is coming from the correct source.

Reply:
I do see what you're saying.. you would basically have two test cases `[InlineData(true, LogLevel.Error)]` and `[InlineData(false, LogLevel.Info)]`, where the second param is the expected level. Would that then make the expected result disjointed from the setup for that expected result (see lines 170 and 178 for example).

I guess i'm just not a huge fan of passing in expected results in test cases as parameters - it kind of goes against the concept of parameterized tests in my personal opinion. My thought process is that the parameters define the test case, not the test result - the result should be determined based on the input data or always be the same regardless.

Reply:
IMO I think it would read a bit easier if I saw this (assume `expectedLogLevel` is the 2nd parameter to the method):
```csharp
logger.Level.Should().Be(expectedLogLevel);
```

Reply:
Crud, I didn't submit my next comment....so I was wondering why it still send "Pending". **Anyway**....

I personally don't have an issue with passing in expected values for the test data that is being passed in as well. Probably coming my personal usage of NUnit with [parameterized tests](https://docs.nunit.org/articles/nunit/writing-tests/attributes/testcase.html). Other examples for xUnit show this as well (e.g. [this one](https://andrewlock.net/creating-parameterised-tests-in-xunit-with-inlinedata-classdata-and-memberdata/)).

Reply:
Yeah, I used NUnit exclusively for years in the past - i had used the `.Returns()` method on `TestCaseData` quite frequently in certain instances - albeit, it was only ever when executing a test that had a very distinct result. For example, writing a test for a method signature `public decimal Multiply(decimal input1, decimal input2)` would have a very explicit and distinct result.

Otherwise, you have to do something like:
```csharp
public decimal SomeTestMethod(decimal input1, decimal input2)
{
   decimal result = SomeFancierMultiplyMethod(input1, input2);
   
   Assert...
   Assert...
   Assert...
   
   return result;
}
```

Anyways, what if I added a line at the top of the test.. something like: `var expectedLogLevel = customLogLevelResolver ? LogLevel.Error : LogLevel.Info` to show that the intention is that the expected log level is dependent on whether or not a custom log level resolver is defined?